extensions/amp-video/0.1/amp-video.js

```diff
@@ -632,11 +660,15 @@ export class AmpVideo extends AMP.BaseElement {
       childElementByTag(this.element, 'video'),
       'Tried to reset amp-video without an underlying <video>.'
     );
-
     this.uninstallEventHandlers_();
     this.installEventHandlers_();
+    if (this.isManagedByBitrate_) {
+      getBitrateManager(this.win).manage(this.video_);
+    }
     // When source changes, video needs to trigger loaded again.
-    this.loadPromise(this.video_).then(() => this.onVideoLoaded_());
+    listenOncePromise(this.video_, 'loadedmetadata').then(() =>
```

Comment:
`this.loadPromise` would not work here since it won't have the sources attached yet, so we use the loadedMetadata directly here.

Reply:
Can you please add a code comment? "Video might not have its sources yet"...

Reply:
You're only handling the case where the video didn't start loading. If the load has started this will never resolve as the `loadedmetadata` event was already fired.

You need to add an extra check on the video readyState:
- If readyState >= 1, onVideoLoaded_()
- otherwise when loadedmetadata, onVideoLoaded()

Reply:
This is for preemptively safety right? I don't think there's a way that readyState can be >=1 here since the `UpdateSourcesTask` is called after this method and we fixed preload Promise.